### ISSUE TITLE
Update builds to Python 3.11

### ISSUE THREAD

**Issue by austinmatherne-wk** [Arelle/Arelle#455]
### What should we change and why?

Python 3.11 was just released. We should update the builds to use 3.11.

**Comment by hermfischer** [comment]
* The new locale.getencoding() may help clean up Locale.py difficulties doing that before.
* The Cython speedup could be useful if we Cythonize formula interpretation loops in XPathContext.py and binding stuff in FormulaEvaluator.py (like the object model in branch dev-cython-xerces)
* math.pow() changes may eliminate special-cased code in formula math library

**Comment by zaneselvans** [comment]
Hello there, we use the Arelle package to extract financial data from the Federal Energy Regulatory Commission's new data system so we can re-publish it in a relational database that's publicly accessible and easier to use for public policymaking.

Do you have any more documentation of the locale difficulties mentioned by @hermfischer? We ran into a hard to debug locale issue earlier this year, in which Arelle was changing the system locale, breaking other parts of our ETL process well downstream (in a totally different Python script, working on a different dataset), and our solution was to fork the Arelle repository (which seemed a bit under-maintained at the time) and just remove the problematic locale change entirely.

If you're familiar with this problem, are there other more appropriate ways to work around it? I've searched the issues here and didn't see any other discussions of locale problems.

**Comment by austinmatherne-wk** [comment]
Hi Zane,

There were locale improvements earlier this year, but the implementation in Arelle still needs some love.

I'm not familiar with the exact issue you ran into, but I'd be happy to work on getting it fixed upstream. Locale handling varies depending on the platform and how Arelle is used (CLI, Python library, GUI). Can you provide more details about what happened and what changes you made to your fork of Arelle to work around it? Opening a bug report issue would be greatly appreciated! :)

**Comment by austinmatherne-wk** [comment]
Update regarding this issue:

Running Arelle from source or a pip install already works on 3.11.

Frozen builds for Python 3.11 are on hold until lxml publishes Python 3.11 Windows wheels. They recently updated their CI (lxml/lxml#360) to build for Windows with 3.11, but they haven't published to PyPI yet. Once that happens we should be able to update our frozen builds to 3.11.

**Comment by zaneselvans** [comment]
I've filed an issue: #505

**Comment by EwoutH** [comment]
Python 3.11 wheels are now available with the release of lxml [4.9.2](https://github.com/lxml/lxml/releases/tag/lxml-4.9.2)!

**Comment by austinmatherne-wk** [comment]
lxml is working. I ran a test build that failed to install cx_Freeze on Windows with Python 3.11. Looks like we're waiting on a wheel from them. In the meantime we can update CI to run our tests/typing/linting on 3.11.
